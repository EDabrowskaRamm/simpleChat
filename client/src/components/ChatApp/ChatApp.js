import React from 'react'

import { ChatAppRoot, Header } from './ChatApp.shards'

import Tabs from '../Tabs/Tabs'

const ChatApp = ({ title = 'Status meeting standup' }) => <ChatAppRoot>
  <Header>{title}</Header>
  <Tabs />
</ChatAppRoot>

export default ChatApp
