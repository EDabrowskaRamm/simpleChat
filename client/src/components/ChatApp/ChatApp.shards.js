import styled from '@emotion/styled'

export const ChatAppRoot = styled.div`
  width: 100%;
  height: 100%;
  overflow: auto;

  @media (min-width: 768px) {
    width: 720px;
    height: 1800px;
  }
`

export const Header = styled.header`
  display: flex;
  align-items: center;
  justify-content: center;
  height: 122px;
  background-color: ${({ theme }) => theme.colors.background.secondary};
`
