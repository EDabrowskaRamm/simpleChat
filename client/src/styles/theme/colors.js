const _white = '#FFF'
const _colorGrey56 = '#868c97'
const _colorGrey61 = '#979c9f'
const _colorGrey63 = '#9a9fa8'
const _colorGrey89 = '#e1e3e5'
const _colorGrey92 = '#ebebeb'

const _pictonBlue = '#3d90f1'
const _oxfordBlue = '#354052'

export const background = {
  primary: _white,
  secondary: _colorGrey92,
}

export const text = {
  primary: _oxfordBlue,
  secondary: _colorGrey63,
  link: _pictonBlue,
  placeholder: _colorGrey61,
  left: _colorGrey56,
}

export const border = {
  input: _colorGrey56,
  chat: _colorGrey89,
  transparent: 'transparent',
}
