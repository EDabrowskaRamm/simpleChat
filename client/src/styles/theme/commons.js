import { border } from './colors'

export const borders = {
  main: `1px solid ${border.chat}`,
  input: `1px solid ${border.input}`,
}
