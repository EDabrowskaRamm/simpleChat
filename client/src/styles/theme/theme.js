import * as colors from './colors'
import * as commons from './commons'

export default {
  colors,
  commons
}
