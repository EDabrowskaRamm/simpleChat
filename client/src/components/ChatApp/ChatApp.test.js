import React from 'react'
import renderer from 'react-test-renderer'
import serializer from 'jest-emotion'
import { ThemeProvider } from 'emotion-theming'
import { Provider } from 'react-redux'
import configureStore from 'redux-mock-store'

import ChatApp from './ChatApp'

import theme from '../../styles/theme/theme'
import initialState from '../../../__mocks__/fileMock'

expect.addSnapshotSerializer(serializer)

describe('renders ChatApp', () => {
  it('Snapshot', () => {
    const mockStore = configureStore()

    const component = renderer.create(
      <Provider store={mockStore(initialState)}>
        <ThemeProvider theme={theme}>
          <ChatApp />
        </ThemeProvider>
      </Provider>)
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
