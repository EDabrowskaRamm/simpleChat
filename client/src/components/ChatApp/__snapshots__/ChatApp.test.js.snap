// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders ChatApp Snapshot 1`] = `
.emotion-12 {
  width: 100%;
  height: 100%;
  overflow: auto;
}

@media (min-width:768px) {
  .emotion-12 {
    width: 720px;
    height: 1800px;
  }
}

.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  height: 122px;
  background-color: #ebebeb;
}

.emotion-5 {
  width: 100%;
}

.emotion-3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: space-around;
  -webkit-justify-content: space-around;
  -ms-flex-pack: space-around;
  justify-content: space-around;
  height: 120px;
  background-color: #ebebeb;
}

.emotion-1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: 50%;
  height: 100%;
  cursor: pointer;
  background-color: #ebebeb;
  border-bottom: 1px solid #e1e3e5;
  background-color: #FFF;
  border-top: 1px solid #e1e3e5;
  border-bottom: 1px solid transparent;
}

.emotion-1:first-of-type {
  border-top-right-radius: 5px;
  border-top-left-radius: 0;
  border-right: 1px solid #e1e3e5;
}

.emotion-1:hover {
  opacity: .9;
}

.emotion-2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: 50%;
  height: 100%;
  cursor: pointer;
  background-color: #ebebeb;
  border-bottom: 1px solid #e1e3e5;
}

.emotion-2:hover {
  opacity: .9;
}

.emotion-6 {
  display: none;
  display: block;
  width: 100%;
}

.emotion-4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
  padding: 10px 40px;
  border-bottom: 1px solid #e1e3e5;
}

.emotion-10 {
  display: none;
}

.emotion-9 {
  width: 100%;
  padding: 10px 40px;
}

.emotion-7 {
  width: 100%;
  padding: 10px 29px;
  border: 1px solid #868c97;
  border-radius: 5px;
}

<div
  className="emotion-12"
>
  <header
    className="emotion-0"
  >
    Status meeting standup
  </header>
  <section
    className="emotion-5"
  >
    <div
      className="emotion-3"
    >
      <p
        className="emotion-1"
        onClick={[Function]}
      >
        Participants 1
      </p>
      <p
        className="emotion-2"
        onClick={[Function]}
      >
        Chat
      </p>
    </div>
    <div>
      <div
        className="emotion-6"
      >
        <section
          className="emotion-5"
        >
          <div
            className="emotion-4"
          >
            <p>
              Jane Downey
            </p>
          </div>
        </section>
      </div>
      <div
        className="emotion-10"
      >
        <section
          className="emotion-9"
        >
          <div>
            <div>
              <div>
                <p>
                  Jane Downey
                </p>
                 
                <span>
                  date
                </span>
              </div>
              <p>
                After losing his wife seven years earlier, the eccentric Dr. John Dolittle, famed doctor and veterinarian of Queen Victoriaâ€™s England, hermits himself away behind the high walls of Dolittle Manor with only his menagerie of exotic animals for company.
              </p>
            </div>
            <div>
              <div>
                <p>
                  Jane Downey
                </p>
                 
                <span>
                  date
                </span>
              </div>
              <p>
                But when the young queen falls gravely ill, a reluctant Dolittle is forced to set sail on an epic adventure to a mythical island in search of a cure, regaining his wit and courage as he crosses old adversaries and discovers wondrous creatures.
              </p>
            </div>
          </div>
          <form
            className="emotion-5"
            onSubmit={[Function]}
          >
            <input
              className="emotion-7"
              id="message"
              onChange={[Function]}
              placeholder="Message"
              type="text"
              value=""
            />
          </form>
        </section>
      </div>
    </div>
  </section>
</div>
`;
